### DISCUSSION TITLE
SQL Functions & Edge Functions

### DISCUSSION THREAD

**Discussion by Falci** [supabase/supabase#31389] [Questions]
I'm trying to do most of the functions as postgres functions, but some are too complex and I implemented as edge functions.
I also enabled the http extension, and I can call the edge function from inside the SQL function (no idea about performance, but this is a pet project, not worried).

My question is: what's the proper way to load the project ID and the API Key?

My first idea was to create a "settings" page with both information.
Is there a better solution out-of-the-box?

**Comment by GaryAustin1** [comment]
Most people use the Vault to store those and then select from the Vault decoded view in their database function.
